gcloud-java-pubsub/src/main/java/com/google/gcloud/pubsub/spi/PublisherSettings.java

```diff
@@ -0,0 +1,333 @@
+import com.google.pubsub.v1.ListTopicSubscriptionsResponse;
+import com.google.pubsub.v1.ListTopicsRequest;
+import com.google.pubsub.v1.ListTopicsResponse;
+import com.google.pubsub.v1.PublishRequest;
+import com.google.pubsub.v1.PublishResponse;
+import com.google.pubsub.v1.PublisherGrpc;
+import com.google.pubsub.v1.Topic;
+import io.grpc.Status;
+
+// Manually-added imports: add custom (non-generated) imports after this point.
+
+// AUTO-GENERATED DOCUMENTATION AND CLASS - see instructions at the top of the file for editing.
+@javax.annotation.Generated("by GAPIC")
+public class PublisherSettings extends ApiCallSettings {
+
+  // =========
+  // Constants
+  // =========
+
+  /**
+   * The default address of the service.
+   *
+   * <!-- manual edit -->
+   * <!-- end manual edit -->
+   */
+  public static final String DEFAULT_SERVICE_ADDRESS = "pubsub-experimental.googleapis.com";
+
+  /**
+   * The default port of the service.
+   *
+   * <!-- manual edit -->
+   * <!-- end manual edit -->
+   */
+  public static final int DEFAULT_SERVICE_PORT = 443;
+
+  /**
+   * The default scopes of the service.
+   */
+  public static ImmutableList<String> DEFAULT_SERVICE_SCOPES =
+      ImmutableList.<String>builder()
+          .add("https://www.googleapis.com/auth/pubsub")
+          .add("https://www.googleapis.com/auth/cloud-platform")
+          .build();
+
+  private static final ImmutableMap<String, ImmutableSet<Status.Code>> RETRYABLE_CODE_DEFINITIONS;
+
+  static {
+    ImmutableMap.Builder<String, ImmutableSet<Status.Code>> definitions = ImmutableMap.builder();
+    definitions.put(
+        "idempotent",
+        Sets.immutableEnumSet(
+            Lists.<Status.Code>newArrayList(
+                Status.Code.DEADLINE_EXCEEDED, Status.Code.UNAVAILABLE)));
+    definitions.put("non_idempotent", Sets.immutableEnumSet(Lists.<Status.Code>newArrayList()));
+    RETRYABLE_CODE_DEFINITIONS = definitions.build();
+  }
+
+  private static final ImmutableMap<String, RetryParams> RETRY_PARAM_DEFINITIONS;
+
+  static {
+    ImmutableMap.Builder<String, RetryParams> definitions = ImmutableMap.builder();
+    RetryParams params = null;
+    params =
+        RetryParams.newBuilder()
+            .setRetryBackoff(
+                BackoffParams.newBuilder()
+                    .setInitialDelayMillis(100L)
+                    .setDelayMultiplier(1.2)
+                    .setMaxDelayMillis(1000L)
+                    .build())
+            .setTimeoutBackoff(
+                BackoffParams.newBuilder()
+                    .setInitialDelayMillis(300L)
+                    .setDelayMultiplier(1.3)
+                    .setMaxDelayMillis(3000L)
+                    .build())
+            .setTotalTimeout(30000L)
+            .build();
+    definitions.put("default", params);
+    RETRY_PARAM_DEFINITIONS = definitions.build();
+  }
+
+  private final ApiCallableBuilder<Topic, Topic> createTopicMethod;
```

Comment:
did you consider changing ApiCallableBuilder (and its similars) not to extend ApiSettings but rather to contain it (member variable). Also, I think ApiCallable could keep a reference to a copy of the settings instead of a copy of individual members.